webapp/src/main/java/org/apache/atlas/web/rest/AuthREST.java

```diff
@@ -199,26 +198,41 @@ private boolean isPolicyUpdated(String serviceName, long lastUpdatedTime) {
         mustClauseList.add(getMap("terms", getMap("typeName", entityUpdateToWatch)));
 
         lastUpdatedTime = lastUpdatedTime == -1 ? 0 : lastUpdatedTime;
-        mustClauseList.add(getMap("range", getMap("timestamp", getMap("gte", lastUpdatedTime))));
+        mustClauseList.add(getMap("range", getMap("created", getMap("gte", lastUpdatedTime))));
 
         dsl.put("query", getMap("bool", getMap("must", mustClauseList)));
 
+        List<Map<String, Object>> sortList = new ArrayList<>();
+        sortList.add(getMap("created", "desc"));
+        dsl.put("sort", sortList);
+
         parameters.setDsl(dsl);
+        Long lastEditTime = 0L; // this timestamp is used to verify if the found policies are synced with any policy create or update op on cassandra
 
         try {
             EntityAuditSearchResult result = auditRepository.searchEvents(parameters.getQueryString());
-
-            if (result == null || CollectionUtils.isEmpty(result.getEntityAudits())) {
-                return false;
+            if (result != null) {
+                if (!CollectionUtils.isEmpty(result.getEntityAudits())) {
+                    EntityAuditEventV2 lastAuditLog = result.getEntityAudits().get(0);
+                    if (!EntityAuditEventV2.EntityAuditActionV2.getDeleteActions().contains(lastAuditLog.getAction()) &&
```

Comment:
In case the event is a delete action, we won't check if we have found the latest policy while fetching policies because the deleted policy would not be there. This is done by keeping the lastEditTime as null.

See below, we won't throw any error if lastEditTime is null. https://github.com/atlanhq/atlas-metastore/pull/3394/files#diff-20486aa08586c2e7a599983d3642dc66652947da2592c37aca0a5546f39d969fR543